File: python/dataStructures/linkedList.py
class Node(object):
    def __init__(self, data: int):
        self.data = data
        self.next = None

class List(object):
    def __init__(self, head: Node):
        self.head = head
    
    def insert(self, node: Node):
        iterator = self.head
        while(iterator is not None and iterator.next is not None):
            iterator = iterator.next
        
        iterator.next = node
    
    def display(self):
        iterator = self.head
        while(iterator is not None):
            print(iterator.data, end="")
            if(iterator.next is not None):
                print(", ", end = "")
            iterator = iterator.next
        print("")
# Initialize a list

myList = List(Node(10))
myList.insert(Node(6))
myList.insert(Node(2))
myList.display()

File: Cpp/linkedList.cpp
#include <iostream>
#include <cstdio>

using namespace std;

struct node {
    int data;
    node *next;
};

class LinkedList{
    private:
    node *head, *tail;
    public:
    LinkedList() {
        head=NULL;
        tail=NULL;
    }

    void createNode(int value) {
        node *temp = new node;
        temp->data = value;
        temp->next = NULL;
        if(head == NULL){
            head = temp;
            tail = temp;
            temp = NULL;
        } else {
            tail->next = temp;
            tail = temp;
        }
    }

    void display() {
        node *temp = new node;
        temp = head;
        while(temp != NULL) {
            cout << temp->data<<"\t";
            temp = temp->next;
        }
    }
};

void createNode(int value);

int main() {
    LinkedList linkedList;
    linkedList.display();
    linkedList.createNode(3);
    linkedList.createNode(5);
    linkedList.display();
    
}